crates/libs/implement/src/lib.rs

```diff
@@ -83,6 +83,25 @@ pub fn implement(attributes: proc_macro::TokenStream, original_type: proc_macro:
         }
     });
 
+    // Dynamic casting requires that the object not contain non-static lifetimes.
+    let enable_dyn_casting = original_type2.generics.lifetimes().count() == 0;
```

Comment:
Is there some way for the end-user to know about this? Can we emit a diagnostic, or at least include this in the public documentation?

Reply:
I'll document the limitation.

Reply:
Documented in the safe cast methods.